stdlib/public/runtime/Metadata.cpp

```diff
@@ -56,19 +62,45 @@
 using namespace swift;
 using namespace metadataimpl;
 
+// allocate memory up to a nearby page boundary
+static void *swift_allocPage(size_t size) {
+#if defined(_MSC_VER)
+  auto mem = VirtualAlloc(
+      nullptr, size, MEM_TOP_DOWN | MEM_RESERVE | MEM_COMMIT, PAGE_READWRITE);
+#else
+  auto mem = mmap(nullptr, size, PROT_READ | PROT_WRITE, MAP_ANON | MAP_PRIVATE,
+                  VM_TAG_FOR_SWIFT_METADATA, 0);
+  if (mem == MAP_FAILED)
+    mem = nullptr;
+#endif
+  return mem;
+}
+
+// free memory allocated by swift_allocPage()
+// On success, swift_freePage() returns 0, on failure -1
+static int swift_freePage(void *addr, size_t size) {
+#if defined(_MSC_VER)
+  return VirtualFree(addr, 0, MEM_RELEASE) == 0 ? -1 : 0;
+#else
+  return munmap(addr, size);
+#endif
+}
+
 void *MetadataAllocator::alloc(size_t size) {
 #if defined(__APPLE__)
   const uintptr_t PageSizeMask = vm_page_mask;
+#elif defined(_MSC_VER)
+  SYSTEM_INFO SystemInfo;
+  GetSystemInfo(&SystemInfo);
+  const uintptr_t PageSizeMask = SystemInfo.dwPageSize - 1;
```

Comment:
Could you factor the page size computation into a helper function, and cache the result there in a static variable?  Then it won't be recomputed every time around.